code/controllers/subsystems/processing/electronics.dm

```diff
@@ -11,86 +11,79 @@
-		new /obj/item/device/electronic_assembly/medium/clam,
-		new /obj/item/device/electronic_assembly/medium/medical,
-		new /obj/item/device/electronic_assembly/medium/gun,
-		new /obj/item/device/electronic_assembly/medium/radio,
-		new /obj/item/device/electronic_assembly/large/default,
-		new /obj/item/device/electronic_assembly/large/scope,
-		new /obj/item/device/electronic_assembly/large/terminal,
-		new /obj/item/device/electronic_assembly/large/arm,
-		new /obj/item/device/electronic_assembly/large/tall,
-		new /obj/item/device/electronic_assembly/large/industrial,
-		new /obj/item/device/electronic_assembly/drone/default,
-		new /obj/item/device/electronic_assembly/drone/arms,
-		new /obj/item/device/electronic_assembly/drone/secbot,
-		new /obj/item/device/electronic_assembly/drone/medbot,
-		new /obj/item/device/electronic_assembly/drone/genbot,
-		new /obj/item/device/electronic_assembly/drone/android,
-		new /obj/item/device/electronic_assembly/wallmount/tiny,
-		new /obj/item/device/electronic_assembly/wallmount/light,
-		new /obj/item/device/electronic_assembly/wallmount,
-		new /obj/item/device/electronic_assembly/wallmount/heavy,
-		new /obj/item/implant/integrated_circuit,
-		new /obj/item/clothing/under/circuitry,
-		new /obj/item/clothing/gloves/circuitry,
-		new /obj/item/clothing/glasses/circuitry,
-		new /obj/item/clothing/shoes/circuitry,
-		new /obj/item/clothing/head/circuitry,
-		new /obj/item/clothing/ears/circuitry,
-		new /obj/item/clothing/suit/circuitry
+	cipherkey = generateRandomString(2000+rand(0,10))
+	circuits_init()
+	..()
+
+/datum/controller/subsystem/processing/electronics/proc/circuits_init()
+	//Cached lists for free performance
+	var/atom/def = /obj/item/integrated_circuit
+	var/default_name = initial(def.name)
+	for(var/path in typesof(/obj/item/integrated_circuit))
+		var/obj/item/integrated_circuit/IC = path
+		var/name = initial(IC.name)
+		if(name == default_name)
+			continue
+		all_integrated_circuits[name] = path // Populating the component lists
+		cached_circuits[IC] = new path()
+		flat_circuit_list += cached_circuits[IC]
+
+		if(!(initial(IC.spawn_flags) & (IC_SPAWN_DEFAULT | IC_SPAWN_RESEARCH)))
+			continue
+
+		var/category = initial(IC.category_text)
+		if(!printer_recipe_list[category])
+			printer_recipe_list[category] = list()
+		var/list/category_list = printer_recipe_list[category]
+		category_list += IC // Populating the fabricator categories
+
+	for(var/path in typesof(/obj/item/device/electronic_assembly))
+		var/obj/item/device/electronic_assembly/A = path
+		var/name = initial(A.name)
+		all_assemblies[name] = path
+		cached_assemblies[A] = new path()
+
+	//printer_recipe_list["Assemblies"] = subtypesof(/obj/item/device/electronic_assembly) - list(/obj/item/device/electronic_assembly/medium, /obj/item/device/electronic_assembly/large, /obj/item/device/electronic_assembly/drone, /obj/item/device/electronic_assembly/wallmount)
```

Comment:
Do we need this comment?

Reply:
A leftover from the time assembly recipe list was redone, will be removed.